flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/CreatingExecutionGraphTest.java

```diff
@@ -149,9 +159,27 @@ public void testSuccessfulSlotAssignmentTransitionsToExecuting() throws Exceptio
                     actualExecutionGraph ->
                             assertThat(actualExecutionGraph, sameInstance(executionGraph)));
 
-            executionGraphWithvertexParallelismFuture.complete(
+            final AtomicReference<Throwable> handledFailureRef = new AtomicReference<>();
+            context.setCreateExecutingStateFunction(
+                    () ->
+                            new AdaptiveSchedulerTest.DummyState() {
+
+                                @Override
+                                public void handleGlobalFailure(Throwable cause) {
+                                    handledFailureRef.set(cause);
+                                }
+                            });
+
+            executionGraphWithVertexParallelismFuture.complete(
                     CreatingExecutionGraph.ExecutionGraphWithVertexParallelism.create(
                             executionGraph, new TestingVertexParallelism()));
+
+            final Throwable t = new RuntimeException("Test.");
+            errorHandlerRef.get().accept(t);
```

Comment:
This should be a separate test case. It now contains more lines for the error handling although that's not it's primary purpose.

Reply:
👍